include/vcpkg/base/files.h

```diff
@@ -318,6 +318,10 @@ namespace vcpkg
         std::vector<std::string> exts;
         bool operator()(const Path& target) const;
     };
+
+#if !defined(_WIN32)
+    void posix_close(int& fd) noexcept;
```

Comment:
I find the `int&` parameter of this public function _very_ spooky, because at the call site this looks too much like `close(x)` -- which I would _not_ expect to modify `x`.

Reply:
Renamed `close_mark_invalid`. If it survives in the first place.